### PR TITLE
Add changes to non-java semantics and to some loops

### PR DESCRIPTION
**Pull request by ivashchenko5**
### **Describe your change:**

- [+] Fix a bug or typo in an existing algorithm?

### **Checklist:**

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] All filenames are in all uppercase characters with no spaces or dashes.
- [x] All functions and variable names follow Java naming conventions.

### CODE REVIEW

File: Sorts/CircleSort.java [hunk 1]
```diff
@@ -9,7 +9,6 @@
     @Override
     public <T extends Comparable<T>> T[] sort(T[] array) {
         int n = array.length;
-        while(doSort(array, 0, n - 1));
```
Comment (siriak) [inline Sorts/CircleSort.java:12]: What's up with that?

---
**Sema Reaction:** :question: I have a question

File: q [hunk 2]
```diff
@@ -0,0 +1,2508 @@
+* [31m9e36810[m -[33m (HEAD -> master, origin/master, origin/HEAD)[m Add surface area of Cone and Hemisphere (#2791) [32m(5 hours ago) [1;34m<Vɪɴᴀʏᴀᴋ Pᴀɴᴅᴇʏ>[m
```
Comment (siriak) [inline q:1]: Please remove that

**Review by siriak — CHANGES_REQUESTED** [review]
@yanglbme @aitorfi what do you think? Should we merge this PR?

---
**Sema Reaction:** :question: I have a question

File: Sorts/CircleSort.java (hunk 1 above)
Comment (aitorfi) [reply Sorts/CircleSort.java:12]: That's not right, you need a loop that calls doSort.

**Review by aitorfi — COMMENTED** [review]
The loop in Sorts/CircleSort can't be removed and I don't think it's necessary to make all these variables final, everything else looks good to me.

**Pull request closed without merging**